backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropCallConvertors.kt

```diff
@@ -151,6 +164,7 @@ private fun InteropCallContext.castPrimitiveIfNeeded(
 }
 
 private fun InteropCallContext.convertEnumToIntegral(enumValue: IrExpression): IrExpression {
+    if (enumValue.type.classOrNull == symbols.nothing) return enumValue
```

Comment:
This is fragile:
```c
enum E {
    ZERO
};

struct S {
    enum E e;
};
```

```kotlin
fun main() {
    foo(E.ZERO)
}

fun <T : E> foo(e: T) {
    val s = nativeHeap.alloc<S>()
    s.e = e
}
```
-> NPE below.

The compiler should always rely on the type of (setter) parameter, not the type of argument.